packages/x-tree-view/src/internals/plugins/useTreeViewLabel/useTreeViewLabel.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import { TreeViewPlugin } from '../../models';
+import { TreeViewItemId } from '../../../models';
+import { UseTreeViewLabelSignature } from './useTreeViewLabel.types';
+
+export const useTreeViewLabel: TreeViewPlugin<UseTreeViewLabelSignature> = ({
+  instance,
+  state,
+  setState,
+  params,
+}) => {
+  const editedItemRef = React.useRef(state.editedItemId);
+
+  const isItemBeingEditedRef = (itemId) => editedItemRef.current === itemId;
+
+  const setEditedItemId = (editedItemId: TreeViewItemId | null) => {
+    setState((prevState) => ({ ...prevState, editedItemId }));
+    editedItemRef.current = editedItemId;
+  };
+
+  const isItemBeingEdited = (itemId: TreeViewItemId) =>
+    itemId === state.editedItemId && editedItemRef.current === itemId;
```

Comment:
I would not do this double check here because we could end up in weird scenarios where the state is not updated, the ref is updated and it returns `false` when it should not.

Does it cause some problems if you just keep:

```suggestion
    itemId === state.editedItemId;
```